portability-core/src/main/java/org/dataportabilityproject/serviceProviders/google/GoogleServiceProvider.java

```diff
@@ -57,16 +62,41 @@
         GmailScopes.GMAIL_MODIFY,
         GmailScopes.GMAIL_LABELS);
 
-    private final GoogleAuth googleAuth;
+    private final ImmutableList<PortableDataType> dataTypes = ImmutableList
+        .of(CALENDAR, MAIL, PHOTOS, TASKS);
+
+    final private Map<PortableDataType, List<String>> dataTypeImportScopes = ImmutableMap.<PortableDataType, List<String>>builder()
+        .put(CALENDAR, Arrays.asList(CalendarScopes.CALENDAR))
+        .put(MAIL, Arrays.asList(GmailScopes.GMAIL_MODIFY))
+        .put(PHOTOS, Arrays.asList("https://picasaweb.google.com/data/"))
+        .put(TASKS, Arrays.asList(TasksScopes.TASKS))
+        .build();
+
+    // Export scopes should only contain READONLY permissions
+    final private Map<PortableDataType, List<String>> dataTypeExportScopes = ImmutableMap.<PortableDataType, List<String>>builder()
+        .put(CALENDAR, Arrays.asList(CalendarScopes.CALENDAR_READONLY))
+        .put(MAIL, Arrays.asList(GmailScopes.GMAIL_READONLY))
+        .put(PHOTOS, Arrays.asList(
+            "https://picasaweb.google.com/data/")) // Photos doesn't have a differing scope for read only.
+        .put(TASKS, Arrays.asList(TasksScopes.TASKS_READONLY))
+        .build();
+
+    private Map<PortableDataType, GoogleAuth> dataTypeImportAuths = new HashMap<>();
+    private Map<PortableDataType, GoogleAuth> dataTypeExportAuths = new HashMap<>();
+
 
     @Inject
     GoogleServiceProvider(AppCredentialFactory appCredentialFactory) throws Exception {
         AppCredentials appCredentials =
             appCredentialFactory.lookupAndCreate("GOOGLE_KEY", "GOOGLE_SECRET");
-        this.googleAuth = new GoogleAuth(
-                appCredentials,
-                // TODO: only use scopes from the products we are accessing.
-                SCOPES);
+
+        for (Entry<PortableDataType, List<String>> e : dataTypeImportScopes.entrySet()) {
```

Comment:
Instead of creating all of them, could we create the right one on demand, e.g. lazy init or users a Suppliers.memoize, as I don't think we need to support multiple exports and imports at the same time.

Reply:
How does this look? I'm not sure if the nested Map<> def is preferred over having two separate maps for import and export from a readability perspective.